### PR TITLE
df: fix calculation of Use% column

### PR DESCRIPTION
**Pull request by cakebaker**
Change formula from: `Used/Size * 100` to `Used/(Used + Avail) * 100`. This formula also works if "Used" and "Avail" do not add up to "Size", which is the case if there are reserved disk blocks.

### CODE REVIEW

File: tests/by-util/test_df.rs [hunk 1]
```diff
@@ -147,11 +147,12 @@ fn test_use_percentage() {
 
     for line in lines {
         let mut iter = line.split_whitespace();
+        // Skip "Filesystem" column
```
Comment (jfinkels) [inline tests/by-util/test_df.rs:150]: Now that the `--output` argument works, let's update this test to just use `--output=used,avail,pcent` so that we don't have to do so much parsing and skipping.

File: src/uu/df/src/table.rs [hunk 2]
```diff
@@ -144,23 +144,24 @@ impl From<Filesystem> for Row {
...
             fs_mount: mount_dir,
             bytes: blocksize * blocks,
-            bytes_used: blocksize * (blocks - bfree),
+            bytes_used: blocksize * bused,
             bytes_avail: blocksize * bavail,
             bytes_usage: if blocks == 0 {
                 None
             } else {
-                Some(((blocks - bfree) as f64) / blocks as f64)
+                Some((bused as f64) / (bused + bavail) as f64)
```
Comment (jfinkels) [inline src/uu/df/src/table.rs:158]: Based on your description, this seems reasonable, but I would not have been able to describe why this change is important without it. Could you please leave a comment in the code here explaining why we should use this formula to compute the usage percentage?

**Review by jfinkels — COMMENTED** [review]
Two minor suggestions. Looks good.

File: tests/by-util/test_df.rs (hunk 1 above)
Comment (cakebaker) [reply tests/by-util/test_df.rs]: Good point, thanks.

File: src/uu/df/src/table.rs (hunk 2 above)
Comment (cakebaker) [reply src/uu/df/src/table.rs:161]: Makes sense, I added a corresponding comment.

File: src/uu/df/src/table.rs [hunk 3]
```diff
@@ -144,23 +144,27 @@ impl From<Filesystem> for Row {
...
-            bytes_used: blocksize * (blocks - bfree),
+            bytes_used: blocksize * bused,
             bytes_avail: blocksize * bavail,
             bytes_usage: if blocks == 0 {
                 None
             } else {
-                Some(((blocks - bfree) as f64) / blocks as f64)
+                // We use "(bused + bavail)" instead of "blocks" because on some filesystems (e.g.
+                // ext4) "blocks" also includes reserved blocks we ignore for the usage calculation.
+                // https://www.gnu.org/software/coreutils/faq/coreutils-faq.html#df-Size-and-Used-and-Available-do-not-add-up
```
Comment (jfinkels) [inline src/uu/df/src/table.rs:160]: A link with more detailed information, excellent!

**Review by jfinkels — APPROVED** [review]
